src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -110,6 +110,113 @@ public struct MetaData
         public int type;
     }
 
+    /// <summary>
+    /// Wrapper over all values emited by the MSIL compiler.
+    /// </summary>
+    internal struct CLRStackValue
+    {
+        public bool IsEnumerable => ArrayUtils.IsEnumerable(Type);
+
+        public bool IsDouble => Value is double;
+
+        public bool IsDefaultArgument => false;
+
+        public bool IsPointer => Type.IsClass;
+
+        public bool IsExplicitCall => false;
+
+        public bool IsNull => Value == null;
+
+        public int Rank // default is 0
+        {
+            get;
+            set;
+        }
+
+        public ProtoCore.Type ProtoType { get; set; }
+
+        public System.Type Type => Value?.GetType();
+
+        public object Value { get; set; }
```

Comment:
Does this mean we need to box value types?

Reply:
yes..we still use boxing/unboxing with this approach